pkg/diagnostics/workflow_metrics.go

```diff
@@ -0,0 +1,143 @@
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+var (
+	elapsedKey = tag.MustNewKey("elapsed")
+
+	reminderTypeKey = tag.MustNewKey("reminder_type")
+
+	executionTypeKey = tag.MustNewKey("execution_type")
+)
+
+const (
+	StatusSuccess   = "success"
+	StatusFailed    = "failed"
+	StatusRetryable = "retryable"
+	CreateWorkflow  = "create_workflow"
+	GetWorkflow     = "get_workflow"
+	AddEvent        = "add_event"
+	PurgeWorkflow   = "purge_workflow"
+	Activity        = "activity"
+	Workflow        = "workflow"
+
+	WorkflowEvent = "event"
+	Timer         = "timer"
+
+	ComponentName = "dapr"
+)
+
+type workflowMetrics struct {
+	// Total Successful and Failed workflow operation requests
+	workflowOperationsCount *stats.Int64Measure
+	// Workflow operation request's Response latency
+	workflowOperationsLatency *stats.Float64Measure
+	// Total workflow/activity reminders created
+	workflowRemindersCount *stats.Int64Measure
+	// Total successful/failed workflow/activity executions
+	workflowExecutionCount *stats.Int64Measure
+	// Time taken to run a workflow to completion
+	workflowExecutionLatency *stats.Float64Measure
+	appID                    string
+	enabled                  bool
+	namespace                string
+}
+
+func newWorkflowMetrics() *workflowMetrics {
+	return &workflowMetrics{
+		workflowOperationsCount: stats.Int64(
+			"runtime/workflow/operations/count",
+			"The number of successful/failed workflow operation requests.",
+			stats.UnitDimensionless),
+
+		workflowOperationsLatency: stats.Float64(
+			"runtime/workflow/operations/latency",
+			"The latencies of responses for workflow operation requests.",
+			stats.UnitMilliseconds),
+		workflowRemindersCount: stats.Int64(
+			"runtime/workflow/reminders/count",
+			"The number of workflow/activity reminders created.",
+			stats.UnitDimensionless),
+		workflowExecutionCount: stats.Int64(
+			"runtime/workflow/execution/count",
+			"The number of successful/failed workflow/activity executions.",
+			stats.UnitDimensionless),
+		workflowExecutionLatency: stats.Float64(
+			"runtime/workflow/execution/latency",
+			"The total time taken to run a workflow/activity to completion.",
+			stats.UnitMilliseconds),
+	}
+}
+
+func (w *workflowMetrics) IsEnabled() bool {
+	return w != nil && w.enabled
+}
+
+// Init registers the workflow metrics views.
+func (w *workflowMetrics) Init(appID, namespace string) error {
+	w.appID = appID
+	w.enabled = true
+	w.namespace = namespace
+
+	return view.Register(
+		diagUtils.NewMeasureView(w.workflowOperationsCount, []tag.Key{appIDKey, componentKey, namespaceKey, operationKey, statusKey}, view.Count()),
+		diagUtils.NewMeasureView(w.workflowOperationsLatency, []tag.Key{appIDKey, componentKey, namespaceKey, operationKey, statusKey}, defaultLatencyDistribution),
+		diagUtils.NewMeasureView(w.workflowRemindersCount, []tag.Key{appIDKey, componentKey, namespaceKey, reminderTypeKey}, view.Count()),
```

Comment:
What is the reason for this metric? Isn't this linearly correlated to the number of operations (1 reminder per each op)? And could users just look at the actor metrics?

Reply:
`workflowRemindersCount` represents the number of internal reminders getting created. It is not linearly correlated to the number of operations, but instead to the reminderName prefixes for the Reminders: https://github.com/dapr/dapr/blob/9e2bcdc3d8a64b02466a9274e9259f166eedc95a/pkg/runtime/wfengine/workflow.go#L632
which basically corresponds to a set of reminders for same work. In total, there are 4 types:1) Workflow 2) Activity 3) Adding Event 4) Timers

It basically gives the total number of reminders getting created, which helps in correlating how the perf etc. is being impacted with remindersCount in workflow.

Reply:
Actually since we already have the reminders Metric: https://github.com/dapr/dapr/blob/9e2bcdc3d8a64b02466a9274e9259f166eedc95a/pkg/diagnostics/service_monitoring.go#L387, this metric doesn't seem to be adding much value. @prateek041 Please remove the reminders metric then...

Reply:
sure